sdk/go/server.go

```diff
@@ -749,6 +344,49 @@ func convertResult(ctx context.Context, result any) (any, error) {
 		return id, nil
 	}
 
+	// TODO: hack, could give CheckResult an ID? Or maybe just need to support "data-only" object fields or something
```

Comment:
Yeah, an id would make this quite trivial. As for "data-only" object fields, do you mean something like #4920?

Reply:
Yes 4920 is exactly what I was thinking of. But overall just making everything IDable should be totally fine too. Really what we care about is being serializable, so whether that's by having the object itself be json serializable or being serializable by converting to an ID doesn't matter.